back/app/services/surveys/average_generator.rb

```diff
@@ -5,7 +5,7 @@ class AverageGenerator
     def initialize(phase, input_type: nil)
       form = phase.custom_form || CustomForm.new(participation_context: phase)
       @fields = IdeaCustomFieldsService.new(form).enabled_fields.select do |f|
-        f.input_type == input_type || f.supports_average? # Defaults to returning all fields that support averages
+        input_type ? f.input_type == input_type : f.supports_average? # Defaults to returning all fields that support averages
       end
```

Comment:
This change introduce different logic around when `supports_average` will be used I think? Just double-checking this change is intended?

Reply:
It is deliberate, because the logic was incorrect before - ie before for each field it would try f.input_type == input_type and otherwise return a field if it supported averages. But what we needed was to only return fields of a certain type if `input_type` was returned